### ISSUE TITLE
Compiler does not check if `id` is constant expression in `subgroupBroadcast`

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/glslang#1427]
Compiler does not check if `id` is constant expression in `subgroupBroadcast`.
This shader succeeds compiling with `glslangValidator.exe -V110 -H --target-env vulkan1.1 .\subgroup.comp`:

```
#version 460

#extension GL_KHR_shader_subgroup_ballot: enable

layout(local_size_x = 128, local_size_y = 1, local_size_z = 1) in;

layout(std430, set=0, binding=0) buffer layout_in_stuff {
  int in_stuff[];
};

void main(){
  int id = in_stuff[0];
  int global_index_b = subgroupBroadcast(42, id);
}
```

**Comment by johnkslang** [comment]
I think this is also supposed to work for, say, a loop inductive variable, and hence is not as simple as checking for a compile-time constant.

**Comment by krOoze** [comment]
Well, the https://github.com/KhronosGroup/GLSL/blob/master/extensions/khr/GL_KHR_shader_subgroup.txt spec says specifically:

> The function subgroupBroadcast() returns the <value> from the invocation
whose <gl_SubgroupInvocationID> is equal to \<id>.  \<id> must be an **integral
constant expression**.

**Comment by johnkslang** [comment]
Ah, it is `subgroupShuffleXor()` that works with loop inductive variables, which is missing. But, will have a look at enforcing `subghroupBroadcast()`, where I thought the proper constant checks had been done.

**Comment by arcady-lunarg** [comment]
This was fixed by #1592 and is a duplicate of #1591.
